edx_rest_framework_extensions/auth/jwt/authentication.py

```diff
@@ -80,11 +97,54 @@ def authenticate(self, request):
             # Errors in production do not need to be logged (as they may be noisy),
             # but debug logging can help quickly resolve issues during development.
             logger.debug('Failed JWT Authentication,', exc_info=exception)
-            # Note: I think this case should only include AuthenticationFailed and PermissionDenied,
-            #   but will monitor the custom attribute to verify.
+            # Useful monitoring for debugging various types of failures.
             set_custom_attribute('jwt_auth_failed', 'Exception:{}'.format(repr(exception)))
```

Comment:
Thoughts on monitoring:
1. Can we add annotations to the various set_custom_attribute calls? (search code base for custom_attribute_name)
2. For the forgiving JWT case, would it be better to rename this to something like jwt_auth_failed_warning? Having one name (no change) may make this simple to find, but it may also be misleading if someone doesn’t know that forgiving jwt custom attributes need to be reviewed as well to know how to interpret.

Reply:
See https://github.com/openedx/edx-drf-extensions/pull/197#pullrequestreview-1572779330 for changes. Marking as resolved.